tests/karate/src/test/java/be/utils/server.feature

```diff
@@ -0,0 +1,75 @@
+@ignore @report=false
+Feature: This feature starts a server and stops it after every scenario.
+
+    Scenario: Start the server and configure Karate
+        # Handler can be used to filter websocket messages 
+        * def handle = function(msg){ karate.signal(msg); return msg.startsWith('{')}
+
+        # Method that waits/pauses for x seconds
+        * def wait = 
+            """
+                function(secs) {
+                    java.lang.Thread.sleep(secs*1000)
+                }
+            """
+
+        # Get the server depending on the environment
+        * def GoServer = Java.type("be.utils.GoServer")
+        * def ScalaServer = Java.type("be.utils.ScalaServer")
+        * def getServer =
+            """
+                function() {
+                    if(env == 'go')
+                        return new GoServer();
+                    else if(env == 'scala')
+                        return new ScalaServer();
+                    else
+                        karate.fail("Unknown environment for server");
+                }
+            """
+        * def server = call getServer
+
+        # Method that waits until host:port is available
+        * def waitForPort = 
+            """
+                function() {
+                    karate.waitForPort(host, port)
+                    // Scala takes more time to start the server
+                    if(env == 'scala')
+                        wait(10)

```

Comment:
this looks very brittle - different systems might have a different timeout.

it might be okay for now though: if the servers start the communication with a `greeting` message, you will be able to wait for that greeting in the future 😄

Reply:
This was needed because scala server takes sometime to startup from sbt(~20 secs). 
The waitForPort(host,port) does a total of 30 polling before failing. This is usually enough for the servers to startup. 
Running the scala backend from  an assembled jar is almost instant .

Reply:
Is the jar assembled after every PR merge or should we assemble it before the tests?